### PR TITLE
challenge 2

### PR DESCRIPTION
**Pull request by me-diru**

### CODE REVIEW

File: challenge-2/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "tag": "week-in-banff"
+    "destination": "Banff",
+    "duration": "1 week",
+    "num_people": "four",
+    "activities": ["hiking", "camping", "long drive", "food"]
+    
+}
+```
+
+You will have to create a prompt out the JSON data given along with the request to feed it to the LLM model to generate the itinerary
```
Comment (karthik2804) [inline challenge-2/README.md:26]:
```suggestion
You will have to create a prompt from the JSON data in the request and feed it to the LLM model to create an itinerary.
```

File: challenge-2/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+PROMPT = " Create a summer vacation detailed itinerary trip to go to <destination> for a <duration>. <num_people> people will be going on this trip planning to do <activities>"
+```
+Do explore different ways to generate the best outcome from the model :D
+
+The expected return type must be a `201` response as follows:
+
+```json
+{
+    "itinerary" : <GENERATED ITINERARY>
+}
```
Comment (karthik2804) [inline challenge-2/README.md:39]: You probably want the backend to generate the tag and not the user providing it to avoid one user overwritting another.

**Review by karthik2804 — COMMENTED** [review]
Looks good overall, I left a couple of suggestions.

File: challenge-2/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+```
+The header type must contain `Content-Type: application/json`
+
+For reference, the users will have to send a `GET` request to route `\<tag>` given along with the previous `POST` request to see the generated itinerary again. Store the itinerary in a KV Store for easier retrieval in `tag:<GENERATED ITINERARY>` format. When requested the response should return with a `200` response code in the following way:
```
Comment (kate-goldenring) [inline challenge-2/README.md:43]: I am having a hard time understanding this section. You seem to be mixing together the user experience description with the backend implementation description. Can you work on rewording it? Maybe move the section about storing the result up to where you talk about the return response, telling them to persist the result before returning 201. Here is a start:
```suggestion
To reference a previously generated itinerary, a user can send a `GET` request to the same endpoint with the  `/<tag>` route. The itinerary is then retrieved from the key-value store where it was cached using its tag. The response should have a `200` response code and be formatted as follows: 
```

File: challenge-2/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+For reference, the users will have to send a `GET` request to route `\<tag>` given along with the previous `POST` request to see the generated itinerary again. Store the itinerary in a KV Store for easier retrieval in `tag:<GENERATED ITINERARY>` format. When requested the response should return with a `200` response code in the following way: 
+
+```json
+{
+    "itinerary": <GENERATED ITINERARY>
+}
+```
+
+Don't forget to put your "spinscreen" on (no pun intended) and all the best! :D
```
Comment (kate-goldenring) [inline challenge-2/README.md:51]: Pun is intended :)
```suggestion
Don't forget to put your "spinscreen" on (pun intended)! Happy vacation! :D
```

**Review by kate-goldenring — COMMENTED** [review]
Made several syntactical suggestions. Thanks for putting this together

File: challenge-2/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# Challenge 2
+Inspired from [Challenge 3 of Advent of Spin 2023](https://github.com/fermyon/advent-of-spin/blob/main/2023/Challenge-3/README.md
+)
+
+
```
[the remark this answers is not in the document]
Comment (me-diru) [reply challenge-2/README.md:5]: Yep! I will remove the inspiration from the challenge 3 part as well.

**Pull request merged**